tests/provision/provisioning_rules.go

```diff
@@ -0,0 +1,1334 @@
+package tests
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	mediaSsd       = "SSD"
+	mediaSata      = "SATA"
+	mediaType      = "media_type"
+	liops          = "iops"
+	llatency       = "latency"
+	pxZone         = "failure-domain.beta.kubernetes.io/px_zone"
+	pxRegion       = "failure-domain.beta.kubernetes.io/px_region"
+	kubeZone       = "failure-domain.beta.kubernetes.io/zone"
+	ruleCategories = 6
+)
+
+type labelDict map[string]interface{}
```

Comment:
reduce the number of lines here to remove the inline yamls and instead create the placement golang struct using helper functions. See how autopilot declares this using in the variable autopilotruleBasicTestCases